src/cert.rs

```diff
@@ -170,7 +153,11 @@ fn remember_extension<'a>(
     if extension.id.len() != ID_CE.len() + 1
         || !extension.id.as_slice_less_safe().starts_with(&ID_CE)
     {
-        return Ok(Understood::No);
+        return if extension.critical {
```

Comment:
Style nit: IMO a `match` for this is slightly easier to read and more concise.

Reply:
Switched to a `match` in the impl of `Extension::unsupported()`.